File: src/main/java/com/woomulwoomul/woomulwoomulbackend/common/log/P6spyLogging.java
package com.woomulwoomul.woomulwoomulbackend.common.log;

import com.p6spy.engine.logging.Category;
import com.p6spy.engine.spy.appender.MessageFormattingStrategy;
import org.hibernate.engine.jdbc.internal.FormatStyle;

public class P6spyLogging implements MessageFormattingStrategy {

    @Override
    public String formatMessage(int connectionId,
                                String now,
                                long elapsed,
                                String category,
                                String prepared,
                                String sql,
                                String url) {
        return formatSql(connectionId, category, sql, elapsed);
    }

    private String formatSql(int connectionId, String category, String sql, long elapsed) {
        String uuid = InterceptorLogging.getRequestId() != null ? InterceptorLogging.getRequestId() : "SYSTEM";
        String sqlLog = "[" + uuid + " | DATABASE] connectionId=" + connectionId + ", time=" + elapsed + "ms";

        if (sql == null || sql.trim().isEmpty())
            return sqlLog;

        if (Category.STATEMENT.getName().equals(category)) {
            String tmpSql = sql.trim().toLowerCase();

            if (tmpSql.startsWith("create")
                    || tmpSql.startsWith("drop")
                    || tmpSql.startsWith("alter")
                    || tmpSql.startsWith("truncate"))
                sqlLog = sqlLog + FormatStyle.DDL.getFormatter().format(sql);
            else
                sqlLog = sqlLog + FormatStyle.BASIC.getFormatter().format(sql);
        }

        return sqlLog;
    }
}

File: core/src/main/kotlin/com/woomulwoomul/core/common/log/P6spyLogging.kt
package com.woomulwoomul.core.common.log

import com.p6spy.engine.logging.Category
import com.p6spy.engine.spy.appender.MessageFormattingStrategy
import org.hibernate.engine.jdbc.internal.FormatStyle

class P6spyLogging : MessageFormattingStrategy {
    override fun formatMessage(
        connectionId: Int,
        now: String?,
        elapsed: Long,
        category: String?,
        prepared: String?,
        sql: String?,
        url: String?,
    ): String {
        val uuid = InterceptorLogging().requestId
        var sqlLog = "[" + uuid + " | DATABASE] connectionId=" + connectionId + ", time=" + elapsed + "ms"

        if (sql == null || sql.trim() == "")
            return sqlLog

        if (Category.STATEMENT.name == category) {
            val tmpSql = sql.trim().lowercase()

            sqlLog += if (tmpSql.startsWith("create") ||
                tmpSql.startsWith("drop") ||
                tmpSql.startsWith("alter") ||
                tmpSql.startsWith("truncate"))
                FormatStyle.DDL.formatter.format(sql)
            else
                FormatStyle.BASIC.formatter.format(sql)
        }

        return sqlLog
    }
}